lib/document.js

```diff
@@ -103,6 +103,12 @@ class Document extends Readable {
     // init destinations name tree
     this._destinations = new PDF.NameTree()
 
+    // init outlines hierarchy
+    this._outlines = []
+    this._outlines[0] = new PDF.Object('Outlines')
+    this._outlines[0].data = {type: 'Outlines'}
+    this._registerObject(this._outlines[0])
```

Comment:
Something you couldn't know, since I haven't documented the internal API. `_registerObject` already reserves an ID for the created object. However, the object is only added if we actually have some outlines. I would therefore recommend calling `this._registerObject(this._outlines[0])` below (I will add another comment there). With this change you might have to revert the commit that updates all tests (since their result will not change anymore).

Reply:
I added the "root outline" to the outlines array to make the chaining to the parents easier ; however the root outline is special (it has a Type property and no siblings so it doesn't need the Prev / Next property). The drawback is that I instantiate the root outline at the doc creation. 

I chose this way because it keeps every indirect reference properties into the outline method.

I will instantiate it the first time the outline method is called. This should answer both concerns.